docs/combinatorics.rst

```diff
@@ -1,6 +1,202 @@
 .. _sec_combinatorics:
 
-=====================================
-Ranking and Unranking Tree Topologies
-=====================================
-TODO
+=============
+Combinatorics
+=============
+This module implements a combinatorial approach to counting topologies
+of leaf-labelled trees. It provides methods for enumerating
+all possible trees with a given number of labelled leaves, as well as
+converting back and forth between a tree and its position, or rank, in
+the enumeration of all possible topologies.
+
++++++++++++++++++++++++++++++++++
+Ranking/Unranking Topologies
++++++++++++++++++++++++++++++++++
+The enumeration of leaf-labelled trees discussed below allows for an
```

Comment:
Which trees? It feels like there used to be a sentence before this. It would be good to briefly discuss what kind of trees we're talking about, I guess.

Reply:
Maybe helpful here to show all the trees of 4 (or something) just as an intuitive way into things?

Reply:
The concept of rank should be clear then.